src/eval-frame/components/reps/data-table-rep.jsx

```diff
@@ -0,0 +1,94 @@
+import React from "react";
+import ReactTable from "react-table";
+import PropTypes from "prop-types";
+import { get } from "lodash";
+import styled from "react-emotion";
+
+import "./react-table-styles.css";
+
+import { tinyRepSerializer } from "./rep-utils/tiny-rep-serializer";
+
+import DefaultRenderer from "./default-handler";
+import tinyRep from "./tiny-reps";
+
+const TableDetails = styled.div`
+  border: solid #e5e5e5;
+  border-width: 0px 1px 1px 1px;
+  padding: 5px;
+`;
+const TableDetailsMessage = styled.div`
+  color: #999;
+  font-style: italic;
+  font-family: "Open Sans", sans-serif;
+  padding-bottom: ${props => (props.pad ? "3px" : "0px")};
+`;
+
+const CellDetails = props => {
+  if (props.focusedPath) {
+    return (
+      <TableDetails>
+        <TableDetailsMessage pad>
+          {`details for ${props.focusedPath}`}
+        </TableDetailsMessage>
+        <DefaultRenderer value={get(props.value, props.focusedPath)} />
+      </TableDetails>
+    );
+  }
+  return (
+    <TableDetails>
+      <TableDetailsMessage>Click a table cell for details</TableDetailsMessage>
+    </TableDetails>
+  );
+};
+
+export default class TableRenderer extends React.Component {
+  static propTypes = {
+    value: PropTypes.any.isRequired
+  };
+  constructor(props) {
+    super(props);
+    this.state = { focusedPath: undefined };
+  }
+
+  handleCellClick(rowInfo, column) {
+    this.setState({ focusedPath: `[${rowInfo.index}]["${column.id}"]` });
+  }
+
+  render() {
+    const { value } = this.props;
+
+    const columns = Object.keys(value[0]).map(k => ({
+      Header: k,
+      accessor: k,
+      Cell: cell => {
+        return tinyRep(tinyRepSerializer(cell.value));
+      }
+    }));
+
+    const pageSize = value.length > 10 ? 10 : value.length;
+    return (
+      <div>
+        <ReactTable
```

Comment:
hmm, noticing I can't resize the columns without everything breaking. Maybe let’s deactivate column resize if we’re going to keep things like this, or fix it.

![Apr-10-2019 13-52-12](https://user-images.githubusercontent.com/95735/55918085-a5733380-5ba6-11e9-81cb-a0cc8630ea2e.gif)

Reply:
ya that's the thing i was mentioning about how it's dumb that react-table is not actually build on an real <table> element, which would let the browser guarantee that the headers always align correctly :-/

i've seen something like this (tho not exactly what your gif shows) happening in the storybook, so i think it might be a problem with react-table rather than an interaction with containing styles, so we might not be able to fix it, in which case i think you're right that disabling column resizing is a good way forward. (and that should be fine for our use case since the columns just show the tiny reps, which don't get too wide)